### ISSUE TITLE
Vue3 can't access toolResponse client side

### ISSUE THREAD

**Issue by cosbgn** [vercel/ai#3066]
### Description

On my backend  I have:

```js
		const result = await streamText({
			model: openai('gpt-4o-mini'),
			system: system_prompt,
			tools: {
				search_files: tool({
					description: `foo`,
					parameters: z.object({
						search_queries: z.array(z.string()).describe('The search query'),
					}),
					execute: async ({ search_queries }) => await search(search_queries)
				})
			},
			maxSteps: 5, // allow up to 5 steps,
			messages
		})
		return result.toDataStreamResponse()
```
On my frontend:

```js
import { useChat } from '@ai-sdk/vue';
const { messages, handleSubmit } = useChat()

```
Tool responses are initially added to the messages but when the callback finished (the message stop streaming) they get removed.

**Comment by 102898Is** [comment]
maybe this will help
https://bit.ly/4gx42eQ

you may need to install the c compiler

**Comment by lgrammel** [comment]
@cosbgn shortened urls like cln.sh are suspicious. please embed the screencast directly here

**Comment by lgrammel** [comment]
Please check out this example https://github.com/vercel/ai/tree/main/examples/nuxt-openai and make sure that you use the latest versions of `@ai-sdk/vue` and `ai`

**Comment by cosbgn** [comment]
Hey @lgrammel thanks for the quick answer. I tested with the example and I get the same results. 

I attach the video directly here, but as a summary `toolInvocations` disappears client side once the streaming ends.

Not sure if related but openAI has a similar bug (https://github.com/openai/openai-node/issues/1060).


https://github.com/user-attachments/assets/262d9075-9b90-467c-aed8-47950ba65a70

**Comment by lgrammel** [comment]
Not related to openai and def strange. What is your frontend code?

**Comment by cosbgn** [comment]
@lgrammel it's identical to the linked example, I only alias the lib name.

```js
<script setup lang="ts">
import { useChat as useAiChat } from '@ai-sdk/vue';
const space = ref({id:123})
const { messages, input, handleSubmit, isLoading, completion, error, data } = useAiChat({ 
	api:`/api/spaces/${space.value.id}/rag/chat_aisdk`,
  	maxSteps: 5,
})
</script>
```

**Comment by lgrammel** [comment]
I have tried reproducing the issue with our standard vue example ( https://github.com/vercel/ai/tree/main/examples/nuxt-openai ) w/o success. The tool results do not disappear. Are you on the latest version of `@ai-sdk/vue` (`0.0.50`) and `@ai` (`3.3.43`)? Can you check if that example works for you ( http://localhost:3000/use-chat-tools )

**Comment by cosbgn** [comment]
@lgrammel I still get the bug on both chome and firefox, could you test this: https://github.com/cosbgn/ai_sdk_bug I really don't understand what it could be.

Here a video of the code running:

https://github.com/user-attachments/assets/bcf08f2e-87a7-4e26-aa56-efa3a82a71bb

**Comment by lgrammel** [comment]
Here you define messages instead of taking them from the request: https://github.com/cosbgn/ai_sdk_bug/blob/main/server/api/chat.js

They should be extract from the request, see e.g.
https://github.com/vercel/ai/blob/main/examples/nuxt-openai/server/api/use-chat-tools.ts

**Comment by cosbgn** [comment]
Hi @lgrammel that was hard-coded to simplify the repo and make sure functions get called, but that isn't the issue. 

I just pushed a new version where I parse the body and the error is definitely still happening.

**Comment by lgrammel** [comment]
Thanks! Was able to reproduce with your example, looks like a bug.

**Comment by lgrammel** [comment]
Identified the issue

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/3081

**Comment by lgrammel** [comment]
releasing `@ai-sdk/vue@0.0.51` with a fix

**Comment by cosbgn** [comment]
Works! Thank you!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This was a concrete failure in existing Vue client behavior, confirmed and fixed by the maintainer.

### Evidence
- In Vue's `useChat`, streamed `toolInvocations` appeared initially but disappeared when streaming completed.
- The maintainer reproduced the behavior and explicitly identified it as a bug.
- A fix was released in `@ai-sdk/vue@0.0.51`, and the reporter confirmed it worked.
